src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -169,38 +170,54 @@ private Collector getHybridQueryCollector() {
      * @return search results that can be reduced be the caller
      */
     @Override
-    public ReduceableSearchResult reduce(Collection<Collector> collectors) {
+    public ReduceableSearchResult reduce(Collection<Collector> collectors) throws IOException {
```

Comment:
Is this exception getting swallowed or thrown from `CollectorManager`?

Reply:
The interface definition says it is thrown: https://github.com/apache/lucene/blob/branch_10_2/lucene/core/src/java/org/apache/lucene/search/CollectorManager.java#L55